### DISCUSSION TITLE
Superimposed Charts

### DISCUSSION THREAD

**Discussion by kaleming** [kernc/backtesting.py#467] [Q&A] [answered]
Hi @kernc,

I found very cool this option you provide to visualize superimposed charts.

I would like to know, if it is possible to accomplish other time frames combinations as well ?  (1 min & 5 min / 3 min & 15 min / 5 min & 15 min). Even if I had to provide `df1` and `df2` already resampled.

**Comment by kernc** [comment, marked as answer]
You could, e.g., [resample original `df`](https://kernc.github.io/backtesting.py/doc/backtesting/lib.html#backtesting.lib.OHLCV_AGG) to `'3T'` (3 min) and pass `bt.plot(..., superimpose='15T')`?

Other than that, you could roll as many as you want of [your own line indicators](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.I):
```py
        ...
        def resample(period, df):
            return df.Open.resample(period).first(), df.Close.resample(period).last()
        self.resampled15t = self.I(resample, '15T', self.data.df, overlay=True)
        ...
```

**Comment by kaleming** [comment]
Hi @kernc, thanks for your reply.

It seems to be working now.

I just have a doubt. 

- I noticed when using 3T with 15T, I can' t visualize only the higher frame when scroll to zoom in, zoom out (in your example with daily data I can see only the higher frame when I zoom in, zoom out). It it possible to modify the colors_lighter parameter to make it easy to visualize the higher frame as well?

**Reply by kernc** [reply]
If you're as zoomed out to not distinguish the superimposed bars, you won't distinguish the _main bars_ if you make the superimposed colors any less light. I think the defaults are reasonable.

You can quite easily change the colors in your own copy, at this place:
https://github.com/kernc/backtesting.py/blob/7671f044d554249b608b3241dd7cca37ec5a35d1/backtesting/_plotting.py#L474-L475

I'd appreciate a screenshot. :stuck_out_tongue_winking_eye:

**Comment by kaleming** [comment]
Hi @kernc, thanks again.

During the zoom in transition using your website example with daily and monthly superimposed data, it appears good as below:

![image_1](https://user-images.githubusercontent.com/51836477/133838274-00ce3eea-5617-437a-b3df-47401d0baed7.png)

When I try to zoom in my 3T / 15T chart, the transition appears like this:

![image_2](https://user-images.githubusercontent.com/51836477/133838535-fddd6357-3c71-47c6-9c61-345d94872dd2.png)

Although both of them works fine after finish zooming transition:

![image_3](https://user-images.githubusercontent.com/51836477/133838781-d1cd056d-6d3c-4c43-bc92-32a8550b57f2.png)

I will try to change some lightness parameters just to see if works fine for my case.

Thanks a lot for your help;

**Reply by kaleming** [reply]
I will probably explore some combination like this:

<img src="https://user-images.githubusercontent.com/51836477/133840424-058211f0-936b-4e78-a718-1cc9e80fc9d1.png" width="700" height="300">

**Reply by kernc** [reply]
> When I try to zoom in my 3T / 15T chart, the transition appears like this:
> 
> ![image_2](https://user-images.githubusercontent.com/51836477/133838535-fddd6357-3c71-47c6-9c61-345d94872dd2.png)
> 
> Although both of them works fine after finish zooming transition

The [level-of-detail change is from Bokeh](https://docs.bokeh.org/en/latest/docs/user_guide/tools.html#controlling-level-of-detail):

> the plot only draws a small fraction of data points during interactive operations (panning or zooming, for example)

**Reply by kaleming** [reply]
Thanks again @kernc .  I didn`t know about these parameters. 
I will have a look on that.
